plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TypeConverter.java

```diff
@@ -179,25 +192,41 @@ private static org.apache.iceberg.types.Type fromDecimal(DecimalType type)
         return Types.DecimalType.of(type.getPrecision(), type.getScale());
     }
 
-    private static org.apache.iceberg.types.Type fromRow(RowType type)
+    private static org.apache.iceberg.types.Type fromRow(RowType type, Optional<ColumnIdentity> columnIdentity, Optional<AtomicInteger> nextFieldId)
     {
         List<Types.NestedField> fields = new ArrayList<>();
-        for (RowType.Field field : type.getFields()) {
+
+        for (int i = 0; i < type.getFields().size(); i++) {
```

Comment:
may be just name this as `fieldIndex`, rather than reassigning to `index` below

Reply:
this because `Variable used in lambda expression should be final or effectively final`